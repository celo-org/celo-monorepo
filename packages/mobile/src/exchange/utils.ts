import BigNumber from 'bignumber.js'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { convertToContractDecimals } from 'src/tokens/saga'
import { getContractKitAsync } from 'src/web3/contracts'

export async function celoTocUsd(celoAmount: BigNumber) {
  const celoAmountInWei = (
    await convertToContractDecimals(celoAmount, CURRENCY_ENUM.GOLD)
  ).integerValue()

  const contractKit = await getContractKitAsync()
  const exchange = await contractKit.contracts.getExchange()
  const exchangeRate = await exchange.getGoldExchangeRate(celoAmountInWei)
  const cUsdAmount = exchangeRate.multipliedBy(celoAmount)

  return cUsdAmount
}
