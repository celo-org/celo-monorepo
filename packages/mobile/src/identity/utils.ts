import BigNumber from 'bignumber.js'

export function isUserBalanceSufficient(
  userBalance: string | number | null,
  estimatedTxAmount: number | string
) {
  if (!userBalance || new BigNumber(userBalance).isLessThan(new BigNumber(estimatedTxAmount))) {
    return false
  }

  return true
}
