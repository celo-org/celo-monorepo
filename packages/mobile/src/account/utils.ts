import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
