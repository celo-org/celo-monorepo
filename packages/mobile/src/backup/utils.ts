import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
