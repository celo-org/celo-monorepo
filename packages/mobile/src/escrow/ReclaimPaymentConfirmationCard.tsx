import HorizontalLine from '@celo/react-components/components/HorizontalLine'
import { MoneyAmount } from '@celo/react-components/components/MoneyAmount'
import PhoneNumberWithFlag from '@celo/react-components/components/PhoneNumberWithFlag'
import colors from '@celo/react-components/styles/colors'
import { fontStyles } from '@celo/react-components/styles/fonts'
import { componentStyles } from '@celo/react-components/styles/styles'
import BigNumber from 'bignumber.js'
import * as React from 'react'
import { withNamespaces, WithNamespaces } from 'react-i18next'
import { StyleSheet, Text, View } from 'react-native'
import { connect } from 'react-redux'
import LineItemRow from 'src/components/LineItemRow'
import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
import { Namespaces } from 'src/i18n'
import Logo from 'src/icons/Logo'
import { RecipientWithContact } from 'src/recipients/recipient'
import { RootState } from 'src/redux/reducers'
import FeeIcon from 'src/send/FeeIcon'
import { getFeeDisplayValue, getMoneyDisplayValue } from 'src/utils/formatting'

export interface OwnProps {
  recipientPhone: string
  recipientContact?: RecipientWithContact
  amount: BigNumber
  fee?: BigNumber
  isLoadingFee?: boolean
  feeError?: Error
  currency: CURRENCY_ENUM
}

interface StateProps {
  defaultCountryCode: string
}

const mapStateToProps = (state: RootState): StateProps => {
  return {
    defaultCountryCode: state.account.defaultCountryCode,
  }
}

type Props = OwnProps & StateProps & WithNamespaces

class ReclaimPaymentConfirmationCard extends React.PureComponent<Props> {
  render() {
    const {
      recipientPhone,
      recipientContact,
      amount,
      fee,
      isLoadingFee,
      feeError,
      currency,
      defaultCountryCode,
      t,
    } = this.props
    const currencySymbol = CURRENCIES[currency].symbol
    const amountWithFees = getMoneyDisplayValue(amount.minus(fee || 0))

    return (
      <View style={[componentStyles.roundedBorder, style.container]}>
        <View style={style.logo}>
          <Logo height={40} />
        </View>
        <MoneyAmount
          symbol={currencySymbol}
          amount={amountWithFees}
          sign={'+'}
          color={colors.celoGreen}
        />
        <HorizontalLine />
        <View style={style.details}>
          {recipientContact && (
            <Text style={[fontStyles.bodySmallSemiBold, style.contactName]}>
              {recipientContact.displayName}
            </Text>
          )}
          {recipientPhone && (
            <PhoneNumberWithFlag
              e164PhoneNumber={recipientPhone}
              defaultCountryCode={defaultCountryCode}
            />
          )}
        </View>
        <View style={style.feeContainer}>
          <LineItemRow
            currencySymbol={currencySymbol}
            amount={amount.toString()}
            title={t('totalSent')}
          />
          <LineItemRow
            currencySymbol={currencySymbol}
            amount={getFeeDisplayValue(fee)}
            title={t('securityFee')}
            titleIcon={<FeeIcon />}
            isLoading={isLoadingFee}
            hasError={!!feeError}
          />
          <LineItemRow
            currencySymbol={currencySymbol}
            amount={amountWithFees}
            title={t('totalRefunded')}
          />
        </View>
      </View>
    )
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    padding: 20,
  },
  feeContainer: {
    marginBottom: 10,
    justifyContent: 'center',
    alignItems: 'stretch',
  },
  contactName: {
    paddingTop: 6,
    textAlign: 'center',
  },
  logo: {
    alignSelf: 'center',
    margin: 'auto',
    marginVertical: 10,
  },
  details: {
    padding: 20,
  },
})

export default connect<StateProps, {}, {}, RootState>(mapStateToProps)(
  withNamespaces(Namespaces.sendFlow7)(ReclaimPaymentConfirmationCard)
)
