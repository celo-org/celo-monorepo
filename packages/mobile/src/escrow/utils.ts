import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
