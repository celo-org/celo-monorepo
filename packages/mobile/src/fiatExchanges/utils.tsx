import { SIMPLEX_URI } from 'src/config'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { navigateToURI } from 'src/utils/linking'

export const openMoonpay = (currencyCode: LocalCurrencyCode, currencyToBuy: CURRENCY_ENUM) => {
  navigate(Screens.MoonPay, {
    localAmount: 0,
    currencyCode,
    currencyToBuy,
  })
}

export const openSimplex = (account: string | null) => {
  navigateToURI(`${SIMPLEX_URI}?address=${account}`)
}
