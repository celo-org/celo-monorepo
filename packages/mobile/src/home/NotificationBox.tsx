import colors from '@celo/react-components/styles/colors'
import variables from '@celo/react-components/styles/variables'
import * as React from 'react'
import { withNamespaces, WithNamespaces } from 'react-i18next'
import { NativeScrollEvent, ScrollView, StyleSheet, View } from 'react-native'
import { connect } from 'react-redux'
import { dismissEarnRewards, dismissInviteFriends, PaymentRequest } from 'src/account'
import { getPaymentRequests } from 'src/account/selectors'
import CeloAnalytics from 'src/analytics/CeloAnalytics'
import { CustomEventNames } from 'src/analytics/constants'
import { componentWithAnalytics } from 'src/analytics/wrapper'
import { PROMOTE_REWARDS_APP } from 'src/config'
import { EscrowedPayment } from 'src/escrow/actions'
import { getReclaimableEscrowPayments } from 'src/escrow/saga'
import { setEducationCompleted as setGoldEducationCompleted } from 'src/goldToken/actions'
import i18n, { Namespaces } from 'src/i18n'
import { backupIcon, homeIcon, inviteFriendsIcon, rewardsAppIcon } from 'src/images/Images'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import EscrowedPaymentReminderNotification from 'src/notifications/EscrowedPaymentReminderNotification'
import PaymentRequestSummaryNotification from 'src/notifications/PaymentRequestSummaryNotification'
import SimpleNotification from 'src/notifications/SimpleNotification'
import { RootState } from 'src/redux/reducers'
import { isBackupTooLate } from 'src/redux/selectors'
import { navigateToVerifierApp } from 'src/utils/linking'

interface StateProps {
  backupCompleted: boolean
  goldEducationCompleted: boolean
  dismissedEarnRewards: boolean
  dismissedInviteFriends: boolean
  paymentRequests: PaymentRequest[]
  backupTooLate: boolean
  sentEscrowPayments: EscrowedPayment[]
}

interface DispatchProps {
  dismissEarnRewards: typeof dismissEarnRewards
  dismissInviteFriends: typeof dismissInviteFriends
  setGoldEducationCompleted: typeof setGoldEducationCompleted
}

type Props = DispatchProps & StateProps & WithNamespaces

const mapStateToProps = (state: RootState): StateProps => ({
  backupCompleted: state.account.backupCompleted,
  goldEducationCompleted: state.goldToken.educationCompleted,
  paymentRequests: getPaymentRequests(state),
  dismissedEarnRewards: state.account.dismissedEarnRewards,
  dismissedInviteFriends: state.account.dismissedInviteFriends,
  backupTooLate: isBackupTooLate(state),
  sentEscrowPayments: state.escrow.sentEscrowedPayments,
})

const mapDispatchToProps = {
  dismissEarnRewards,
  dismissInviteFriends,
  setGoldEducationCompleted,
}

interface State {
  currentIndex: number
}

export class NotificationBox extends React.Component<Props, State> {
  state = {
    currentIndex: 0,
  }

  escrowedPaymentReminderNotification = () => {
    return getReclaimableEscrowPayments(this.props.sentEscrowPayments).map((payment) => (
      <EscrowedPaymentReminderNotification key={payment.paymentID} payment={payment} />
    ))
  }

  paymentRequestsNotification = (): Array<React.ReactElement<any>> => {
    const { paymentRequests } = this.props
    if (paymentRequests && paymentRequests.length) {
      return [<PaymentRequestSummaryNotification key={1} requests={paymentRequests} />]
    }
    return []
  }

  generalNotifications = (): Array<React.ReactElement<any>> => {
    const {
      t,
      backupCompleted,
      goldEducationCompleted,
      dismissedEarnRewards,
      dismissedInviteFriends,
    } = this.props
    const actions = []

    if (!backupCompleted) {
      actions.push({
        title: t('backupKeyFlow6:yourBackupKey'),
        text: t('backupKeyFlow6:backupKeyNotification'),
        image: backupIcon,
        ctaList: [
          {
            text: t('backupKeyFlow6:getBackupKey'),
            onPress: () => {
              CeloAnalytics.track(CustomEventNames.get_backup_key)
              navigate(Screens.BackupIntroduction)
            },
          },
        ],
      })
    }

    if (!dismissedEarnRewards && PROMOTE_REWARDS_APP) {
      actions.push({
        title: i18n.t('walletFlow5:earnCeloRewards'),
        text: i18n.t('walletFlow5:earnCeloGold'),
        image: rewardsAppIcon,
        ctaList: [
          {
            text: i18n.t('walletFlow5:startEarning'),
            onPress: () => {
              this.props.dismissEarnRewards()
              CeloAnalytics.track(CustomEventNames.celorewards_notification_confirm)
              navigateToVerifierApp()
            },
          },
          {
            text: t('maybeLater'),
            onPress: () => {
              this.props.dismissEarnRewards()
              CeloAnalytics.track(CustomEventNames.celorewards_notification_dismiss)
            },
          },
        ],
      })
    }

    if (!goldEducationCompleted) {
      actions.push({
        title: t('global:celoGold'),
        text: i18n.t('exchangeFlow9:whatIsGold'),
        image: homeIcon,
        ctaList: [
          {
            text: t('exchange'),
            onPress: () => {
              this.props.setGoldEducationCompleted()
              CeloAnalytics.track(CustomEventNames.celogold_notification_confirm)
              navigate(Screens.ExchangeHomeScreen)
            },
          },
          {
            text: t('maybeLater'),
            onPress: () => {
              this.props.setGoldEducationCompleted()
              CeloAnalytics.track(CustomEventNames.celogold_notification_dismiss)
            },
          },
        ],
      })
    }

    if (!dismissedInviteFriends) {
      actions.push({
        title: i18n.t('inviteFlow11:inviteFriendsToCelo'),
        text: i18n.t('inviteFlow11:inviteAnyone'),
        image: inviteFriendsIcon,
        ctaList: [
          {
            text: i18n.t('global:inviteFriends'),
            onPress: () => {
              this.props.dismissInviteFriends()
              CeloAnalytics.track(CustomEventNames.invitefriends_notification_confirm)
              navigate(Screens.Invite)
            },
          },
          {
            text: t('maybeLater'),
            onPress: () => {
              this.props.dismissInviteFriends()
              CeloAnalytics.track(CustomEventNames.invitefriends_notification_dismiss)
            },
          },
        ],
      })
    }

    return actions.map((notification, i) => <SimpleNotification key={i} {...notification} />)
  }

  paginationDots = (notifications: Array<React.ReactElement<any>>) => {
    if (notifications.length < 2) {
      return null
    }
    return (
      <View style={styles.pagination}>
        {notifications.map((n, i) => {
          return (
            <View
              key={i}
              style={this.state.currentIndex === i ? activeDotStyle : passiveDotStyle}
            />
          )
        })}
      </View>
    )
  }

  handleScroll = (event: { nativeEvent: NativeScrollEvent }) => {
    this.setState({
      currentIndex: Math.round(event.nativeEvent.contentOffset.x / variables.width),
    })
  }

  render() {
    const notifications = [
      ...this.paymentRequestsNotification(),
      ...this.escrowedPaymentReminderNotification(),
      ...this.generalNotifications(),
    ]

    if (!notifications || !notifications.length) {
      // No notifications, no slider
      return null
    }
    return (
      <View style={styles.body}>
        <ScrollView
          horizontal={true}
          pagingEnabled={true}
          showsHorizontalScrollIndicator={false}
          // @ts-ignore TODO(cmcewen): should be fixed with new RN types
          onScroll={this.handleScroll}
        >
          {notifications.map((notification, i) => (
            <View key={i} style={styles.notificationContainer}>
              {notification}
            </View>
          ))}
        </ScrollView>
        {this.paginationDots(notifications)}
      </View>
    )
  }
}

const PROGRESS_CIRCLE_PASSIVE_SIZE = 6
const PROGRESS_CIRCLE_ACTIVE_SIZE = 8

const styles = StyleSheet.create({
  body: {
    maxWidth: variables.width,
    width: variables.width,
  },
  notificationContainer: {
    width: variables.width - 2 * variables.contentPadding,
    margin: variables.contentPadding,
  },
  pagination: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    paddingBottom: variables.contentPadding,
    alignItems: 'center',
  },
  circle: {
    flex: 0,
    borderRadius: 8,
    marginHorizontal: 5,
  },
  circlePassive: {
    backgroundColor: colors.inactive,
    height: PROGRESS_CIRCLE_PASSIVE_SIZE,
    width: PROGRESS_CIRCLE_PASSIVE_SIZE,
  },
  circleActive: {
    backgroundColor: colors.celoGreen,
    height: PROGRESS_CIRCLE_ACTIVE_SIZE,
    width: PROGRESS_CIRCLE_ACTIVE_SIZE,
  },
})

const activeDotStyle = StyleSheet.flatten([styles.circle, styles.circleActive])
const passiveDotStyle = StyleSheet.flatten([styles.circle, styles.circlePassive])

export default componentWithAnalytics(
  connect<StateProps, DispatchProps, {}, RootState>(
    mapStateToProps,
    mapDispatchToProps
  )(withNamespaces(Namespaces.walletFlow5)(NotificationBox))
)
