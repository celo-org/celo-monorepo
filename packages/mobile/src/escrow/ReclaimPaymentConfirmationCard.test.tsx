import { CURRENCY_ENUM } from '@celo/utils/src/currencies'
import BigNumber from 'bignumber.js'
import * as React from 'react'
import { Provider } from 'react-redux'
import * as renderer from 'react-test-renderer'
import ReclaimPaymentConfirmationCard from 'src/escrow/ReclaimPaymentConfirmationCard'
import {} from 'src/home/NotificationBox'
import { createMockStore } from 'test/utils'
import { mockE164Number, mockRecipient } from 'test/values'

const store = createMockStore()

jest.mock('src/web3/contracts', () => ({
  isZeroSyncMode: jest.fn().mockReturnValueOnce(false),
}))

describe('ReclaimPaymentConfirmationCard', () => {
  it('renders correctly for send payment confirmation', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ReclaimPaymentConfirmationCard
          recipientPhone={mockE164Number}
          recipientContact={mockRecipient}
          amount={new BigNumber(10)}
          currency={CURRENCY_ENUM.DOLLAR}
          fee={new BigNumber(0.01)}
        />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
