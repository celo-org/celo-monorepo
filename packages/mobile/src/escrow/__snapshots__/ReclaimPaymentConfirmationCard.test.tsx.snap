// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReclaimPaymentConfirmationCard renders correctly for send payment confirmation 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#EEEEEE",
        "borderRadius": 3,
        "borderWidth": 1,
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "margin": "auto",
        "marginVertical": 10,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../packages/mobile/src/images/coins-logo.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "marginHorizontal": 25,
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Hind-Regular",
            "fontSize": 34,
          },
          Object {
            "color": "#42D689",
          },
        ]
      }
    >
      +
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#81868B",
            "fontFamily": "Hind-Regular",
            "fontSize": 24,
            "textAlignVertical": "top",
          },
          Object {
            "color": "#42D689",
          },
        ]
      }
    >
      $
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#81868B",
            "fontFamily": "Hind-Regular",
            "fontSize": 48,
            "lineHeight": 64,
          },
          Object {
            "color": "#42D689",
          },
        ]
      }
    >
      9.99
    </Text>
  </View>
  <View
    style={
      Object {
        "borderStyle": "solid",
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 10,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-SemiBold",
            "fontSize": 14,
            "lineHeight": 18,
          },
          Object {
            "paddingTop": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      John Doe
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Light",
              "fontSize": 14,
              "textAlignVertical": "center",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#EEEEEE",
              "borderRadius": 3,
              "justifyContent": "center",
              "marginRight": 5,
              "paddingLeft": 6,
              "paddingRight": 3,
            },
          ]
        }
      >
        ðŸ‡ºðŸ‡¸ +1
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Light",
            "fontSize": 14,
            "textAlignVertical": "center",
          }
        }
      >
        (415) 555-0000
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "justifyContent": "center",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            }
          }
        >
          totalSent
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        10
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            }
          }
        >
          securityFee
        </Text>
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          isTVSelectable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
              "paddingHorizontal": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/mobile/src/images/info.png",
              }
            }
            style={
              Object {
                "height": 12,
                "tintColor": "#42D689",
                "width": 12,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        0.01
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            }
          }
        >
          totalRefunded
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        9.99
      </Text>
    </View>
  </View>
</View>
`;
